src/odig.ml

```diff
@@ -0,0 +1,62 @@
+open Import
+
+type t = Opam.t * Opam.Switch.t
+
+type of_opam_error = Odig_not_installed
+
+let of_opam (opam, switch) =
+  let open Promise.Syntax in
+  let+ is_installed = Opam.has_package opam switch "odig" in
+  if is_installed then
+    Ok (opam, switch)
+  else
+    Error Odig_not_installed
+
+let cmd_ouput (opam, switch) args =
+  let cmd = Opam.exec opam switch ~args in
+  Cmd.output cmd
+
+let conf t key =
+  let open Promise.Syntax in
+  let+ odig_conf_result = cmd_ouput t [ "odig"; "conf" ] in
+  match odig_conf_result with
+  | Error _ -> None
+  | Ok conf ->
+    let lines = String.split_lines conf in
+    let is_whitespace_or_single_quote c =
+      Char.equal c '\'' || Char.is_whitespace c
+    in
+    List.map lines ~f:(String.rsplit2 ~on:':')
+    |> List.filter_opt
+    |> List.find_map ~f:(fun (k, v) ->
+           if String.equal k key then
+             Some
+               (String.strip v ~drop:is_whitespace_or_single_quote
+               |> Path.of_string)
+           else
+             None)
+
+let cache_dir t =
+  let opam, switch = t in
+  let default_cache_dir = Path.(Opam.path opam switch / "/var/cache/odig/") in
+  let open Promise.Syntax in
+  let+ cache_dir = conf t "cache-dir" in
```

Comment:
Good point. As long as we only need the `cache_dir` config it's probably better since it simplifies the code. https://github.com/ocamllabs/vscode-ocaml-platform/pull/771/commits/c4620028547d15a004867a25f138af189e676357